### PR TITLE
Merge 2024-10-14

### PR DESCRIPTION
**Pull request by silabs-robin**
This PR merges from `core-v-verif` to `cv32e40s-dv`.
That is, from branch `cv32e40s/dev`, taking the `git subtree` of `core-v-verif/cv32e40s/`.
_(The goal is to do the same as for the x, to delete the dir in cvverif and have a *-dv repo instead.)_

--------

Method:
Approximately same as in https://github.com/openhwgroup/cv32e40s-dv/pull/2

--------

Test status:
* Hello world:  ✅  (Passed)
* ci_check:  ✅  (All passed, `CV_CORE=cv32e40s ./bin/ci_check --iss 0 -s xrun`)
* Formal (s-dv):  ✅  (Compiles and runs, but CEXes)
* Formal (internal):  ✅  (Compiles and runs, no CEXes in a few minutes)

Cloned tb with:
`CV_CORE=cv32e40s VERIF_ENV_REPO=https://github.com/silabs-robin/cv32e40s-dv.git  VERIF_ENV_REF=cv32e40s_subtree_cv32e40sdev_2024-10-14  ./bin/clonetb  --clone`

Note: Ran without ISS, because I haven't had my idv license server configured in a while.

--------

How to review.

Compare old commits:
See the [latest commits of this repo](https://github.com/openhwgroup/cv32e40s-dv/commits/main/), "Aug 17, 2023".
See the [first commits of this PR](https://github.com/openhwgroup/cv32e40s-dv/pull/3/commits), "Aug 18, 2023".
^ Indicates that this PRs diff, along with "git subtree", has picked up at the right location in history.

Compare new commits:
See the [latest commits of core-v-verif](https://github.com/openhwgroup/core-v-verif/commits/cv32e40s/dev/cv32e40s), "correct indent again".
See the [latest commits of this PR](https://github.com/openhwgroup/cv32e40s-dv/pull/3/commits), "correct indent again".
^ Indicates that "git subtree" has gathered commits correctly.

Consider the "Test status" results.

Look at the "Files changed", if you want, but there are so many lines changed that I don't think it is feasible.
Rely instead on comparing the commits.

If one wants to run tests, checkout core-v-verif at cv32e40s/dev, then use `clonetb` as described above.

### CODE REVIEW

**Comment by silabs-robin** [comment]
Hi, @MikeOpenHWGroup, I have reappeared :)

@MikeOpenHWGroup, @silabs-hfegran, and @silabs-krdosvik, I don't have permissions to add reviewers directly on this repo, but I would like you all to have a look at this PR if you would.

(I also likely don't have the power to merge, since I can't add reviewers.)

File: docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv
Comment (silabs-hfegran) [inline docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv:337]: These, and the above seems to be incorrect for 40s - no atomic support

**Comment by silabs-krdosvik** [comment]
Looks good to me

**Review by silabs-krdosvik — APPROVED** [review]
Looks good

File: docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv
Comment (silabs-robin) [reply docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv:337]: Seems incorrect in what sense?
If those lines are incorrect in the sense that the vplan content should be written differently, then that is feedback for https://github.com/openhwgroup/core-v-verif/pull/2284
If those lines are incorrect in the sense that this merge went wrong, then that is feedback for this PR.

But it seems this content is how [that file looks](https://github.com/openhwgroup/core-v-verif/blob/cv32e40s/dev/cv32e40s/docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv?plain=1#L335) in `core-v-verif/cv32e40s/` on branch `cv32e40s/dev`, so unless I am mistaken I can't see that this indicates any problem with the merge.

Note also that the file is prefixed with `CV32E40SX_`, as these vplans apparently were shared for S and X.

Comment (silabs-hfegran) [reply docs/VerifPlans/Simulation/privileged_spec/CV32E40SX_PMA_VerifPlan.csv:337]: I do not want to block the merge with this, but it felt weird to me that N/A for 40S got changed to some cover that is only applicable to 40X.

**Comment by MikeOpenHWGroup** [comment]
>Hi, @MikeOpenHWGroup, I have reappeared :)

Fantastic news!  I am now back from my "summer sabbatical", so the timing is good.

**Review by silabs-krdosvik — APPROVED** [review]
Looks good.

**Review by silabs-hfegran — APPROVED** [review]

File: bsp/bsp.h
Comment (silabs-robin) [inline bsp/bsp.h:1]: Hi @MikeOpenHWGroup.

Both Henrik and Kristine have given their approvals, but they "don't count", because none of us have commiter(?) privileges on this repo.
We also can't merge (because none of us have commiter(?) privileges on this repo).

* Would you also like some time to review this PR?
* Could we get the same commiter(?) privileges on this repo as on core-v-verif?

Comment (MikeOpenHWGroup) [reply bsp/bsp.h:1]: > Would you also like some time to review this PR?

LGTM, so I will approve.  I'll leave the merge to you so that you get credit (or blame!) as the author.

> Could we get the same commiter(?) privileges on this repo as on core-v-verif?

Yes, this should be possible.  We will need to transfer this repo to the [Eclipse CORE-V](https://projects.eclipse.org/projects/openhw.corev) project.  As you probably do not know, your Committer priviledges only extend to repos under the CORE-V project and not all OpenHW repos are part of CORE-V.  There is a process for this which (hopefully?) painless, and I'll get started on that.

**Review by MikeOpenHWGroup — APPROVED** [review]

File: bsp/bsp.h
Comment (silabs-robin) [reply bsp/bsp.h:1]: > LGTM, so I will approve

Thank you 🙇

>  There is a process for [transfering to Eclipse CORE-V] which (hopefully?) painless, and I'll get started on that.

Yes, please. Thanks. That would be nice.

>  I'll leave the merge to you so that you get credit (or blame!) as the author.

I can take the credit/blame :)
However, I will have to wait for the CORE-V transfer, as my powers are currently limited. (If it only takes 1-2 days it is not much of a problem. We can just wait a bit and see.)

**Pull request merged**
